source/modulo_components/modulo_components/utilities/utilities.py

```diff
@@ -12,3 +15,37 @@ def parse_topic_name(topic_name: str) -> str:
     sanitized_string = re.sub("\W", "", topic_name, flags=re.ASCII).lower()
     sanitized_string = sanitized_string.lstrip("_")
     return sanitized_string
+
+
+def modify_parameter_overrides(parameter_overrides: List[Parameter]) -> List[Parameter]:
+    """
+    Modify parameter overrides to handle the rate and period parameters.
+    This function checks for existence of the rate and period parameter in the parameter overrides and modifies them to
+    correspond to the same value. If both the rate and the period parameter exist, the period will be set from the rate.
+
+    :param parameter_overrides: The parameter overrides passed to the component constructor
+    :return: The modified parameter overrides 
+    """
+    rate = None
+    period = None
+    parameters = []
+    for parameter in parameter_overrides:
+        if parameter.name == "rate":
+            rate = parameter
+        elif parameter.name == "period":
+            period = parameter
+        else:
+            parameters.append(parameter)
+    
+    if rate is not None and period is not None:
```

Comment:
remind me, is there no way to check "NOT_SET" parameters in python?

Reply:
I think we could do the same as in C++ yes, if you prefer